upstairs/src/lib.rs

```diff
@@ -4016,9 +4348,13 @@ async fn up_listen(
                                         "Timed out for up.guest.recv drain, \
                                         returning gracefully"
                                     );
-                                    return;
                                 }
                             }
+                            // XXX Do we break here?
+                            // what do I do?  Exit?
+                            // Maybe all current downstairs need to go back
+                            // to NEW?
+                            panic!("Write more code");
```

Comment:
This is the graceful shutdown path (entered when we dropped `ds_status_tx` in the `Do not attempt to reconnect if we were marked inactive` code) which means we should have returned above?

Reply:
The drop of ds_status_tx and exiting of looper was causing a problem if a downstairs connected then disconnected before we went active.

To support the graceful shutdown I think there is a bit more work to do.  This is already a big set of changes and I'm wondering if it is enough to polish this and do the full support for graceful shutdown in the next PR.

Reply:
:+1: